src/preamble.js

```diff
@@ -605,12 +605,20 @@ function abort(what) {
   // Use a wasm runtime error, because a JS error might be seen as a foreign
   // exception, which means we'd run destructors on it. We need the error to
   // simply make the program stop.
+  // FIXME This is not working now, because Wasm EH currently does not assume
+  // all RuntimeErrors are from traps; it decides whether a RuntimeError is from
+  // a trap or not based on a hidden field within the object. So at the moment
+  // we don't have a way of throwing a trap from JS. TODO Make a JS API that
+  // allows this.
```

Comment:
```suggestion
  // allows this in the wasm spec.
```
(just to clarify the TODO is not emscripten-specific)